Tests/TestUtilities/Assertions/TestLoggerAssertions.cs

```diff
@@ -0,0 +1,264 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using FluentAssertions.Execution;
+using FluentAssertions.Primitives;
+
+namespace TestUtilities.Assertions;
+
+public static class TestLoggerExtensions
+{
+    public static TestLoggerAssertions Should(this TestLogger subject) =>
+        new(subject);
+}
+
+public class TestLoggerAssertions : ReferenceTypeAssertions<TestLogger, TestLoggerAssertions>
+{
+    protected override string Identifier { get; } = nameof(TestLogger);
+
+    public TestLoggerAssertions(TestLogger subject) : base(subject) { }
+
+    [CustomAssertion]
+    public AndConstraint<TestLoggerAssertions> HaveErrors(int expectedCount, string because = "", params string[] becauseArgs)
```

Comment:
Consider using `AndWhichConstraint` and return `Subject.Errors` as `Which` (This makes chaining easier).